hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ContainerKeyScanner.java

```diff
@@ -0,0 +1,421 @@
+                    long start = System.currentTimeMillis();
+                    directoryTable = getDirectoryTableData(parent.getDbPath());
+                    long end = System.currentTimeMillis();
+                    out().println(
+                        "directoryTable loaded in " + (end - start) + " ms.");
+                    isDirTableLoaded = true;
+                  }
+                  keyName.append(getFsoKeyPrefix(volumeId, bucketId, value));
+                }
+                keyName.append(value.getKeyName());
+                containerKeyInfos.add(
+                    new ContainerKeyInfo(locationInfo.getContainerID(),
+                        value.getVolumeName(), volumeId, value.getBucketName(),
+                        bucketId, keyName.toString(),
+                        value.getParentObjectID()));
+              }
+            }
+          }
+        }
+        iterator.get().next();
+        keysProcessed++;
+      }
+      return keysProcessed;
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String removeBeginningSlash(String path) {
+    if (path.startsWith(OM_KEY_PREFIX)) {
+      return path.substring(1);
+    }
+
+    return path;
+  }
+
+  private String getFsoKeyPrefix(long volumeId, long bucketId,
+                                 OmKeyInfo value) {
+    String prefix =
+        OM_KEY_PREFIX + volumeId + OM_KEY_PREFIX + bucketId +
+            OM_KEY_PREFIX;
+    Set<Long> dirObjIds = new HashSet<>();
+    dirObjIds.add(value.getParentObjectID());
+    Map<Long, Path> absolutePaths =
+        getAbsolutePathForObjectIDs(bucketId, prefix, Optional.of(dirObjIds));
+    Path path = absolutePaths.get(value.getParentObjectID());
+    String keyPath;
+    if (path.toString().equals(OM_KEY_PREFIX)) {
+      keyPath = path.toString();
+    } else {
+      keyPath = path + OM_KEY_PREFIX;
+    }
+
+    return removeBeginningSlash(keyPath);
+  }
+
+
+  private ColumnFamilyHandle getColumnFamilyHandle(
+      byte[] name, List<ColumnFamilyHandle> columnFamilyHandles) {
+    return columnFamilyHandles
+        .stream()
+        .filter(
+            handle -> {
+              try {
+                return Arrays.equals(handle.getName(), name);
+              } catch (Exception ex) {
+                throw new RuntimeException(ex);
+              }
+            })
+        .findAny()
+        .orElse(null);
+  }
+
+  private String removeTrailingSlashIfNeeded(String dbPath) {
+    if (dbPath.endsWith(OzoneConsts.OZONE_URI_DELIMITER)) {
+      dbPath = dbPath.substring(0, dbPath.length() - 1);
+    }
+    return dbPath;
+  }
+
+  private void printOutput(ContainerKeyInfoWrapper containerKeyInfoWrapper) {
+    List<ContainerKeyInfo> containerKeyInfos =
+        containerKeyInfoWrapper.getContainerKeyInfos();
+    if (containerKeyInfos.isEmpty()) {
+      out().println("No keys were found for container IDs: " + containerIds);
+      out().println(
```

Comment:
For json tools we should only print json to stdout and print everything else to stderr otherwise it will break when piping to tools like `jq`.

Reply:
Done.